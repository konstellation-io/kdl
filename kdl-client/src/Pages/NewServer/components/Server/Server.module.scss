@import 'Styles/colors';
@import 'Styles/mixins';
@import 'Styles/variables';

@function get-triangle-center($triangle-size, $inner-triangle-size) {
  @return (4 * $triangle-size - $inner-triangle-size) / 6;
}

$size-t1: 24px;
$size-t2: 172px;
$size-t3: 216px;
$size-t4: 250px;
.triangle {
  position: relative;
  font-family: 'Montserrat', sans-serif;

  user-select: none;
  height: $size-t4;
  width: $size-t4;

  transition: transform ease 0.3s;

  &.down {
    transform: rotateX(180deg);
  }

  div {
    @include center;
    pointer-events: none;
  }

  .t3,
  .t4 {
    transition: font-size ease 0.3s, text-shadow ease 0.3s;
  }

  .t1 {
    font-size: $size-t1;
    top: get-triangle-center($size-t4, $size-t1);
    color: palette(base);
    -webkit-text-stroke: 1px palette(highlight);
    text-shadow: 0 0 10px palette(highlight, 800);

    transition: font-size ease 0.3s, text-shadow ease 0.3s;
  }

  .t2 {
    font-size: $size-t2;
    top: get-triangle-center($size-t4, $size-t2);
    color: transparent;
    background-color: palette(base, 900);
    background-clip: text;
    text-shadow: 0 0 4px palette(base, 700);

    transition: font-size ease 0.2s, background-color ease 0.3s;
  }
  .t3 {
    font-size: $size-t3;
    top: get-triangle-center($size-t4, $size-t3);
    color: transparent;
    -webkit-text-stroke: 1px palette(highlight);
    filter: drop-shadow(0px 0px 2px palette(highlight));
  }
  .t4 {
    font-size: $size-t4;
    top: get-triangle-center($size-t4, $size-t4);
    color: transparent;
    -webkit-text-stroke: 1px palette(feedback, 700);
    filter: drop-shadow(0px 0px 4px palette(highlight));
  }
}

.triangle:hover,
.hoverContainer:hover {
  .triangle {
    transform: scale(1.1);

    &.down {
      transform: scale(1.1) rotateX(180deg);
    }
  }

  .t4 {
    font-size: $size-t4 * 1.1;
  }
  .t3 {
    font-size: $size-t3 * 1.05;
  }
  .t2 {
    font-size: $size-t2 * 0.98;
    background-color: palette(highlight);
  }
  .t1 {
    text-shadow: 0 0 10px #fff, 0 0 20px palette(highlight),
      0 0 30px palette(highlight);
  }
}
