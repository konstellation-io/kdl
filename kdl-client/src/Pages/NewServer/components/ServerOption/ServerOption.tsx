import { Button } from 'kwc';
import { Link } from 'react-router-dom';
import React from 'react';
import containerStyles from '../../NewServer.module.scss';
import cx from 'classnames';
import serverStyles from '../Server/Server.module.scss';
import styles from './ServerOption.module.scss';

type Props = {
  title: string;
  subtitle: string;
  actionLabel: string;
  to: string;
  Server: JSX.Element;
};

function ServerOption({ title, subtitle, actionLabel, to, Server }: Props) {
  return (
    <Link to={to}>
      <div
        className={cx(
          styles.container,
          serverStyles.hoverContainer,
          containerStyles.server
        )}
      >
        <div>{Server}</div>
        <p className={styles.title}>{title}</p>
        <p className={styles.subtitle}>{subtitle}</p>
        <div className={styles.button}>
          <Button label={actionLabel} primary />
        </div>
      </div>
    </Link>
  );
}

export default ServerOption;
