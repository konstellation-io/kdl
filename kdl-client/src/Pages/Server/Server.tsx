import { useParams } from 'react-router-dom';
import { RouteServerParams } from 'Constants/routes';
import useServers from '../../Hooks/useServers';
import React, { useEffect } from 'react';
import { ipcRenderer } from 'electron';

function Server() {
  const { serverId } = useParams<RouteServerParams>();
  const { getServer } = useServers();
  const server = getServer(serverId);

  useEffect(() => () => ipcRenderer.send('closeServer'), []);

  useEffect(() => {
    ipcRenderer.send('closeServer');
    console.log(server?.url);
    // FIXME: pass the admin-ui url as arg in the send function
    ipcRenderer.send('loadServer');
  }, [serverId]);

  return null;
}

export default Server;
