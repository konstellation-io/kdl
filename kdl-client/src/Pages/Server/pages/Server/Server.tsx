import ROUTE, { RouteServerParams } from 'Constants/routes';
import React, { useEffect } from 'react';
import { Route, Switch, useParams } from 'react-router-dom';

import Project from './pages/Project/Project';
import Projects from './pages/Projects/Projects';
import ServerBar from './components/ServerBar/ServerBar';
import UserSshKey from './pages/UserSshKey/UserSshKey';
import UserApiTokens from './pages/UserApiToken/UserApiTokens';
import Users from './pages/Users/Users';
import styles from './Server.module.scss';
import { titlebar } from 'Components/TitleBar/TitleBar';
import useOpenedServer from '../../apollo/hooks/useOpenedServer';
import useServers from 'Hooks/useServers';

function Server() {
  const { serverId } = useParams<RouteServerParams>();
  const { getServer } = useServers();
  const server = getServer(serverId);

  const { updateOpenedServer } = useOpenedServer();

  useEffect(() => {
    if (server) {
      updateOpenedServer(server);
      titlebar.updateTitle(`Konstellation - ${server.name}`);
    }
  }, [server, updateOpenedServer]);

  useEffect(
    () => () => {
      updateOpenedServer(null);
      titlebar.updateTitle('Konstellation');
    },
    // We want to execute this on on component mount/unmount
    // eslint-disable-next-line react-hooks/exhaustive-deps
    []
  );

  return (
    <div className={styles.container}>
      <ServerBar />
      <Switch>
        <Route exact path={ROUTE.SERVER_USERS} component={Users} />
        <Route exact path={ROUTE.SERVER} component={Projects} />
        <Route exact path={ROUTE.PROJECT} component={Project} />
        <Route exact path={ROUTE.USER_SSH_KEY} component={UserSshKey} />
        <Route exact path={ROUTE.USER_API_TOKENS} component={UserApiTokens} />
      </Switch>
    </div>
  );
}

export default Server;
