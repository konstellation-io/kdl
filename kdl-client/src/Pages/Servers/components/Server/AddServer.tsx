import IconAdd from '@material-ui/icons/Add';
import React from 'react';
import styles from './Server.module.scss';
import { useHistory } from 'react-router-dom';

type Props = {
  label: string;
  to: string;
};
function AddServer({ label, to }: Props) {
  const history = useHistory();

  return (
    <div className={styles.addServerContainer}>
      <div className={styles.addServerBg} onClick={() => history.push(to)} />
      <div className={styles.addServerLabel}>
        <IconAdd className="icon-regular" />
        <p>{label}</p>
      </div>
    </div>
  );
}

export default AddServer;
