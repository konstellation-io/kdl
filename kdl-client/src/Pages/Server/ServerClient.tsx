import { ApolloClient, ApolloProvider } from '@apollo/client';
import React, { useEffect, useRef } from 'react';
import { Route, Switch } from 'react-router-dom';

import NewProject from './pages/NewProject/NewProject';
import NewUser from './pages/NewUser/NewUser';
import ROUTE from 'Constants/routes';
import Server from './pages/Server/Server';
import cache from './apollo/cache';

function ServerClient() {
  // Resets cache on exit client
  useEffect(
    () => () => {
      cache.reset();
    },
    []
  );

  const client = useRef(
    new ApolloClient({
      uri: 'http://localhost:4000/graphql',
      credentials: 'include',
      cache,
    })
  );

  return (
    <ApolloProvider client={client.current}>
      <Switch>
        <Route exact path={ROUTE.NEW_PROJECT} component={NewProject} />
        <Route exact path={ROUTE.NEW_SERVER_USER} component={NewUser} />
        <Route path={ROUTE.SERVER} component={Server} />
      </Switch>
    </ApolloProvider>
  );
}

export default ServerClient;
