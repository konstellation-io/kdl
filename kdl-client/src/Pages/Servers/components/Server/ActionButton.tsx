import React, { MouseEvent } from 'react';

import { Action } from './Server';
import styles from './Server.module.scss';
import { useHistory } from 'react-router-dom';

function ActionButton({ label, Icon, onClick, to }: Action) {
  const history = useHistory();

  function handleOnClick(event?: MouseEvent<HTMLDivElement>) {
    event?.stopPropagation();
    event?.preventDefault();

    if (onClick) onClick();
    else if (to) history.push(to || '');
  }

  return (
    <div className={styles.actionButtonContainer}>
      <div onClick={handleOnClick} className={styles.actionButtonBg} />
      <div className={styles.actionButtonContent}>
        <Icon className={'icon-regular'} />
        <p className={styles.actionButtonLabel}>{label}</p>
      </div>
    </div>
  );
}

export default ActionButton;
