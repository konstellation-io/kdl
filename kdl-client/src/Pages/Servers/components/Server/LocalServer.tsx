import ROUTE, { buildRoute } from 'Constants/routes';
import Server, { ServerBaseProps } from './Server';

import React from 'react';
import useServers from 'Hooks/useServers';

export enum LocalServerStates {
  STARTED = 'STARTED',
  STOPPED = 'STOPPED',
  STARTING = 'STARTING',
  STOPPING = 'STOPPING',
}

export interface LocalServerProps extends ServerBaseProps {
  state: LocalServerStates;
}
export function LocalServer(props: LocalServerProps) {
  const { getServerActions } = useServers();
  const actions = getServerActions(props.state, props.serverId);

  const serverReady = props.state === LocalServerStates.STARTED;
  const onOpenUrl = serverReady
    ? buildRoute.server(ROUTE.SERVER, props.serverId)
    : null;

  return <Server {...props} actions={actions} onOpenUrl={onOpenUrl} local />;
}

export default LocalServer;
