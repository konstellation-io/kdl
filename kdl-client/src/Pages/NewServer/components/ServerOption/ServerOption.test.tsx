import Server, { DIRECTION } from '../Server/Server';

import React from 'react';
import ServerOption from './ServerOption';
import { shallow } from 'enzyme';

const component = shallow(
  <ServerOption
    title="Some title"
    subtitle="Some subtitle"
    actionLabel="Some label"
    to="/some/route"
    Server={<Server direction={DIRECTION.UP} />}
  />
);

describe('ServerOption component', () => {
  test('Component match snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('show right texts', () => {
    expect(component.contains('Some title')).toBeTruthy();
    expect(component.contains('Some subtitle')).toBeTruthy();
  });

  it('show right components', () => {
    expect(component.find(Server).isEmptyRender()).toBeFalsy();
  });
});
