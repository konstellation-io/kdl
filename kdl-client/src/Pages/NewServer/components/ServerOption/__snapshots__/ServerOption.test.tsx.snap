// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServerOption component Component match snapshot 1`] = `
<Link
  to="/some/route"
>
  <div
    className="container hoverContainer server"
  >
    <div>
      <Server
        direction="up"
      />
    </div>
    <p
      className="title"
    >
      Some title
    </p>
    <p
      className="subtitle"
    >
      Some subtitle
    </p>
    <div
      className="button"
    >
      <P
        label="Some label"
        primary={true}
      />
    </div>
  </div>
</Link>
`;
